### PR TITLE
Update response model

### PR DESCRIPTION
**Pull request by mtrojanowski**

### CODE REVIEW

File: src/main/java/io/curity/authenticator/netid/client/NetIdAccessAuthenticationAttributes.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ private static SubjectAttributes getSubjectAttributes(ResultCollect result)
     private static void handleSubject(UserInfoType userInfo, List<Attribute> subjectAttributes)
     {
         @Nullable String personalNumber = userInfo.getPersonalNumber();
-        @Nullable String userId = userInfo.getUserId();
```
Comment (mtrojanowski) [inline src/main/java/io/curity/authenticator/netid/client/NetIdAccessAuthenticationAttributes.java:106]: The `userId` is still present in the WSDL so we could fall back to this one also. We can also fall back to the `requestedUserId` field, as it seems that it is the only one that will be in the response for sure. Any thoughts?

Comment (daniellindau) [reply src/main/java/io/curity/authenticator/netid/client/NetIdAccessAuthenticationAttributes.java:106]: We need to find a subject to use, and as long as all the attributes we find is available in the result the admin can change to the one they want using an action. To not break previous functinoality, I suggest we do a prio-list:
`personalNumber -> userId -> uniqueUserName -> requestedUserId`, and then put all four of them in the result. Wdyt?

File: src/main/resources/wsdl/service-server.wsdl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<wsdl:definitions xmlns:tm="http://microsoft.com/wsdl/mime/textMatching/" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:mime="http://schemas.xmlsoap.org/wsdl/mime/" xmlns:tns="http://netid.secmaker.com/nias/" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:s="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://schemas.xmlsoap.org/wsdl/soap12/" xmlns:http="http://schemas.xmlsoap.org/wsdl/http/" targetNamespace="http://netid.secmaker.com/nias/" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/">
+<wsdl:definitions xmlns:s="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://schemas.xmlsoap.org/wsdl/soap12/" xmlns:http="http://schemas.xmlsoap.org/wsdl/http/" xmlns:mime="http://schemas.xmlsoap.org/wsdl/mime/" xmlns:tns="http://netid.secmaker.com/nias/" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:tm="http://microsoft.com/wsdl/mime/textMatching/" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" targetNamespace="http://netid.secmaker.com/nias/" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/">
```
Comment (daniellindau) [inline src/main/resources/wsdl/service-server.wsdl:2]: Does this mean that this version is only compatible with this version of the nias server? My SOAP is rusty

**Review by daniellindau — COMMENTED** [review]
Comment on how to create subject attributes.

File: src/main/java/io/curity/authenticator/netid/client/NetIdAccessAuthenticationAttributes.java (hunk 1 above)
Comment (mtrojanowski) [reply src/main/java/io/curity/authenticator/netid/client/NetIdAccessAuthenticationAttributes.java:106]: Ok, so we populate the subject according to the prio-list, then add all four as subject parameters to the result, right? It seems like a good approach.

File: src/main/resources/wsdl/service-server.wsdl (hunk 2 above)
Comment (daniellindau) [reply src/main/resources/wsdl/service-server.wsdl:2]: I ment that we update the wsdl, but as long as we don't use `nivResponse` we should be backwards compatible. But I don't see usages of that.

**Review by ju-cu — APPROVED** [review]

**Review by daniellindau — APPROVED** [review]

**Comment by ju-cu** [comment]
Agree. That's a clean solution. Add a separate attribute for the
userUniqueName and prioritize accordingly. No need to include the
requestedUserId in the subject, though. The requestedUserId is either the
personalNumber or userUniqueName thus already covered by the other
attributes.

**Pull request merged**